src/pybind/mgr/orchestrator.py

```diff
@@ -1339,7 +1340,7 @@ def validate_add(self):
 
 class NFSServiceSpec(ServiceSpec):
     def __init__(self, name, pool=None, namespace=None, placement=None):
-        super(NFSServiceSpec, self).__init__(name, placement)
+        super(NFSServiceSpec, self).__init__('nfs', name, placement)
 
         #: RADOS pool where NFS client recovery data is stored.
         self.pool = pool
```

Comment:
Hm, this makes it hard to deserialize `NFSServiceSpec`s. 

Let's have a yaml

```yaml
service_type: nfs
name: mynfs
pool: mypool
namespace: mynamespace
```

how can we write a function

```python
class ServiceSpec:
    @classmethod
    from_yaml(input: str): ...
```
that returns a `NFSServiceSpec`? I'm a bit against creating a `ServiceSpecFactory`